Log error message in CSI VSC when in error state

anshulahuja98 opened the pull request:
Fixes: https://github.com/vmware-tanzu/velero/issues/5566

blackpiglet commented on internal/util/util.go:
```diff
@@ -175,6 +175,9 @@ func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1api.VolumeSnap
 		// bound to the existing snapshot.
 		if snapshotContent.Status == nil || snapshotContent.Status.SnapshotHandle == nil {
 			log.Infof("Waiting for volumesnapshotcontents %s to have snapshot handle. Retrying in %ds", snapshotContent.Name, interval/time.Second)
+			if snapshotContent.Status == nil && snapshotContent.Status.Error != nil {
```
Looks like the if condition check should be 
`if snapshotContent.Status != nil && snapshotContent.Status.Error != nil`

blackpiglet commented on internal/util/util.go:
```diff
@@ -183,7 +186,11 @@ func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1api.VolumeSnap
 	if err != nil {
 		if err == wait.ErrWaitTimeout {
-			log.Errorf("Timed out awaiting reconciliation of volumesnapshot %s/%s", volSnap.Namespace, volSnap.Name)
+			if snapshotContent.Status == nil && snapshotContent.Status.Error != nil {
```
Looks like the if condition check should be
`if snapshotContent.Status == nil && snapshotContent.Status.Error != nil`

anshulahuja98 replied:
Resolved

anshulahuja98 replied:
!= as per other message resolved

blackpiglet commented:
Could you squash the 3 commits into one?

anshulahuja98 commented:
> Could you squash the 3 commits into one? The DCO check failure is due to no signoff in the commit message. You can resolve this by adding `--signoff` parameter in the `git commit` command.

- Squashed the commits
- Signed off the commit
- Added changelog
